Tests/xSharePoint/xSharePoint.xSPAddInManagementServiceApp.Tests.ps1

```diff
@@ -0,0 +1,120 @@
+                Set-TargetResource @testParams
+                Assert-MockCalled New-SPAppManagementServiceApplication
+            }
+        }
+
+        Context "When service applications exist in the current farm but the specific Add-in app does not" {
+
+            Mock Get-SPServiceApplication { return @(@{
+                TypeName = "Some other service app type"
+            }) }
+
+            It "returns null from the Get method" {
+                Get-TargetResource @testParams | Should BeNullOrEmpty
+                Assert-MockCalled Get-SPServiceApplication -ParameterFilter { $Name -eq $testParams.Name } 
+            }
+
+        }
+
+        Context "When a service application exists and is configured correctly" {
+            Mock Get-SPServiceApplication { 
+                return @(@{
+                    TypeName = "App Management Service Application"
+                    DisplayName = $testParams.Name
+                    ApplicationPool = @{ Name = $testParams.ApplicationPool }
+                    Database = @{
+                        Name = $testParams.DatabaseName
+                        Server = @{ Name = $testParams.DatabaseServer }
+                    }
+                })
+            }
+
+            It "returns values from the get method" {
+                Get-TargetResource @testParams | Should Not BeNullOrEmpty
+                Assert-MockCalled Get-SPServiceApplication -ParameterFilter { $Name -eq $testParams.Name } 
+            }
+
+            It "returns true when the Test method is called" {
+                Test-TargetResource @testParams | Should Be $true
+            }
+        }
+
+        Context "When a service application exists and the app pool is not configured correctly" {
+            Mock Get-SPServiceApplication { 
+                $service = @(@{
+                    TypeName = "App Management Service Application"
+                    DisplayName = $testParams.Name
+                    ApplicationPool = @{ Name = "Wrong App Pool Name" }
+                    Database = @{
+                        Name = $testParams.DatabaseName
+                        Server = @{ Name = $testParams.DatabaseServer }
+                    }
+                })
+                    
+                $service = $service | Add-Member ScriptMethod Update {
+                    $Global:xSPAppServiceUpdateCalled = $true
+                } -PassThru 
+                return $service
+            }
+            Mock Get-SPServiceApplicationPool { 
+                @{ Name = $testParams.ApplicationPool } }
+
+            It "returns false when the Test method is called" {
+                Test-TargetResource @testParams | Should Be $false
+            }
+
+            It "calls the update service app cmdlet from the set method" {
+                Set-TargetResource @testParams
+                Assert-MockCalled Get-SPServiceApplicationPool
+                $Global:xSPAppServiceUpdateCalled | Should Be $true
```

Comment:
Can you set this global to be false before the method call? That way if we execute the tests twice in one window (locally that is) we wont see false positives